pkg/cluster/cluster.go

```diff
@@ -255,16 +256,35 @@ func setOptionsDefaults(options Options) Options {
 // NewClientFunc allows a user to define how to create a client.
 type NewClientFunc func(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error)
 
-// DefaultNewClient creates the default caching client.
+// ClientOptions are the optional arguments for tuning the caching client.
+type ClientOptions struct {
+	UncachedObjects   []client.Object
+	CacheUnstructured bool
+}
+
+// DefaultNewClient creates the default caching client, that will never cache Unstructured.
 func DefaultNewClient(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
-	c, err := client.New(config, options)
-	if err != nil {
-		return nil, err
-	}
+	return ClientBuilderWithOptions(ClientOptions{})(cache, config, options, uncachedObjects...)
+}
 
-	return client.NewDelegatingClient(client.NewDelegatingClientInput{
-		CacheReader:     cache,
-		Client:          c,
-		UncachedObjects: uncachedObjects,
-	})
+// ClientBuilderWithOptions returns a Client constructor that will build a client
+// honoring the options argument
+func ClientBuilderWithOptions(options ClientOptions) NewClientFunc {
+	return func(cache cache.Cache, config *rest.Config, clientOpts client.Options, uncachedObjects ...client.Object) (client.Client, error) {
+		if uncachedObjects != nil {
+			options.UncachedObjects = append(options.UncachedObjects, uncachedObjects...)
+		}
```

Comment:
No reason for the conditional:

```suggestion
		options.UncachedObjects = append(options.UncachedObjects, uncachedObjects...)
```

Reply:
Thx! 😊